artemis/modules/drupal_scanner.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional
+
+import bs4
+from karton.core import Task
+
+from artemis import http_requests
+from artemis.binds import TaskStatus, TaskType, WebApplication
+from artemis.modules.base.base_newer_version_comparer import (
+    BaseNewerVersionComparerModule,
+)
+
+
+class DrupalScanner(BaseNewerVersionComparerModule):
+    """
+    Drupal scanner - checks whether the version is obsolete.
+    """
+
+    identity = "drupal_scanner"
+    filters = [
+        {"type": TaskType.WEBAPP.value, "webapp": WebApplication.DRUPAL.value},
+    ]
+    software_name = "drupal"
+
+    # Some homepages are big - let's override the default downloaded content size because
+    # we want to identify Drupal version based on a script which is at the bottom.
+    DOWNLOADED_CONTENT_PREFIX_SIZE = 5 * 1024 * 1024
+
+    def run(self, current_task: Task) -> None:
+        url = current_task.get_payload("url")
+        response = http_requests.get(url, max_size=DrupalScanner.DOWNLOADED_CONTENT_PREFIX_SIZE)
+        soup = bs4.BeautifulSoup(response.content)
+
+        version: Optional[str] = None
+        for script in soup.findAll("script"):
+            if "/core/misc/drupal.js?v=" in script.get("src", ""):
+                version = script["src"].split("=")[1]
+
+        if version:
+            is_version_obsolete = super().is_version_obsolete(
+                version,
+            )
+        else:
+            is_version_obsolete = None
+        result = {
+            "version": version,
+            "is_version_obsolete": is_version_obsolete,
+        }
+
+        if version and is_version_obsolete:
+            found_problems = [f"Drupal version {version} on {url} is obsolete"]
+        elif not version:
+            found_problems = [f"Unable to obtain Drupal version for {url}"]
```

Comment:
not sure if it is something to log, it always can be not a drupal app

Reply:
made this an exception as this may mean that the check is broken. This karton takes only identified Drupal apps

Reply:
made this an exception as this may mean that the check is broken. This karton takes only identified Drupal apps